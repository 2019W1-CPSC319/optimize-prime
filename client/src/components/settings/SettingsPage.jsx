import React, { Component } from 'react';
import './settingsPage.css';
import {
  TextField,
  Paper,
  Typography,
  Button
} from '@material-ui/core';
import Swal from 'sweetalert2';

const swalWithBootstrapButtons = Swal.mixin({
  customClass: {
    confirmButton: 'btn btn-success',
    cancelButton: 'btn btn-danger'
  },
  buttonsStyling: true,
});

class SettingsPage extends Component {
  constructor(props) {
    super(props);
    this.state = { subject: '', body: '', signature: '' };
  }

  async componentDidMount() {
    const { actions } = this.props;
    try {
      await actions.getEmailTemplate();
    } catch (error) {
      console.log(error);
    }
    const { template: { subject, body, signature } } = this.props;
    this.setState({ subject, body, signature });
  }

  onChangeTextField = (fieldKey, event) => {
    this.setState({ [fieldKey]: event.target.value });
  }

  onSubmit = async () => {
    const { actions } = this.props;
    const { subject, body, signature } = this.state;
    await actions.updateEmailTemplate(subject, body, signature);
    swalWithBootstrapButtons.fire(
      'Change has been saved!',
      'You\'re all set.',
      'success'
    );
  }

  render() {
    const { subject, body, signature } = this.state;
    return (
      <div>
        <h1 className="settings-title">Settings</h1>
        <Paper style={{ margin: '30px', padding: '30px' }}>
          <Typography
            variant='h6'
          >Email Template</Typography>
          <TextField
            label="Subject"
            onChange={(event) => this.onChangeTextField('subject', event)}
            margin="normal"
            value={subject}
            variant="outlined"
            style={{ width: '500px', display: 'block' }}
            inputProps={{
              style: { width: '500px', display: 'block' }
            }}
          />
          <TextField
            label="Body"
            multiline
            onChange={(event) => this.onChangeTextField('body', event)}
            margin="normal"
            value={body}
            variant="outlined"
            style={{ width: '500px', display: 'block' }}
            inputProps={{
              style: { width: '500px', display: 'block' }
            }}
          />
          <TextField
            label="Signature"
            multiline
            onChange={(event) => this.onChangeTextField('signature', event)}
            margin="normal"
            value={signature}
            variant="outlined"
            style={{ width: '500px', display: 'block' }}
            inputProps={{
              style: { width: '500px', display: 'block' }
            }}
          />
          <Button
            variant='outlined'
            style={{
              padding: '10px 30px',
              marginTop: '10px'
            }}
            onClick={this.onSubmit}
          >
            Save
          </Button>
        </Paper>
      </div >
    )
  }
}

export default SettingsPage;
