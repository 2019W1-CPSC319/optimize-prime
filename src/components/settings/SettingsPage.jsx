import React, { Component } from 'react';

class SettingsPage extends Component {
  constructor(props) {
    super(props);
    this.state = {

    };
  }

  render() {
    return <div>SETTINGS PAGE</div>;
  }
}

export default SettingsPage;
