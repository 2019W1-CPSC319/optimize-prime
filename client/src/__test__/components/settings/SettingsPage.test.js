import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SettingsPage from '../../../components/settings/SettingsPage';

describe('Settings Page - Render', () => {
  let wrapper;
  const classes = {};
  const actions = {
    getEmailTemplate: jest.fn(),
  };
  const template = {
    subject: 'Interview with Galvanize',
    body: 'Please provide your availability.',
    signature: 'Galvanize HR Staff',
  };

  beforeEach(() => {
    wrapper = shallow(<SettingsPage
      classes={classes}
      actions={actions}
      template={template}
    />);
  });

  it('render the component', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
