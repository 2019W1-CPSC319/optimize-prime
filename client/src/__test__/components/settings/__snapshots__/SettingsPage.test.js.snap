// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Page - Render match snapshot 1`] = `
<div>
  <h1
    className="settings-title"
  >
    Settings
  </h1>
  <WithStyles(ForwardRef(Paper))
    style={
      Object {
        "margin": "30px",
        "padding": "30px",
      }
    }
  >
    <WithStyles(ForwardRef(Typography))
      variant="h6"
    >
      Email Template
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(TextField))
      inputProps={
        Object {
          "style": Object {
            "display": "block",
            "width": "500px",
          },
        }
      }
      label="Subject"
      margin="normal"
      onChange={[Function]}
      style={
        Object {
          "display": "block",
          "width": "500px",
        }
      }
      value="Interview with Galvanize"
      variant="outlined"
    />
    <WithStyles(ForwardRef(TextField))
      inputProps={
        Object {
          "style": Object {
            "display": "block",
            "width": "500px",
          },
        }
      }
      label="Body"
      margin="normal"
      multiline={true}
      onChange={[Function]}
      style={
        Object {
          "display": "block",
          "width": "500px",
        }
      }
      value="Please provide your availability."
      variant="outlined"
    />
    <WithStyles(ForwardRef(TextField))
      inputProps={
        Object {
          "style": Object {
            "display": "block",
            "width": "500px",
          },
        }
      }
      label="Signature"
      margin="normal"
      multiline={true}
      onChange={[Function]}
      style={
        Object {
          "display": "block",
          "width": "500px",
        }
      }
      value="Galvanize HR Staff"
      variant="outlined"
    />
    <WithStyles(ForwardRef(Button))
      onClick={[Function]}
      style={
        Object {
          "marginTop": "10px",
          "padding": "10px 30px",
        }
      }
      variant="outlined"
    >
      Save
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Paper))>
</div>
`;
